From c40b423b1cec0d54e84e75099b97c022455121ab Mon Sep 17 00:00:00 2001
From: Jonathan Bell <jonathan@raspberrypi.com>
Date: Wed, 15 May 2024 15:21:14 +0100
Subject: [PATCH 1089/1135] DTS: bcm2712: set nonzero QoS values for PCIE1

If PCIE1 is left with the default (zero) AXI QoS values, endpoints can
receive extremely poor service for non-posted transactions e.g. reads.
Such transactions can take milliseconds to complete on a contended
system.

Bump priorities for every TC above the non-realtime greedy peripherals
in BCM2712, to allow reasonable service without competing against hard
realtime peripherals.

Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
---
 arch/arm/boot/dts/broadcom/bcm2712-rpi-5-b.dts | 4 ++++
 1 file changed, 4 insertions(+)

--- a/arch/arm/boot/dts/broadcom/bcm2712-rpi-5-b.dts
+++ b/arch/arm/boot/dts/broadcom/bcm2712-rpi-5-b.dts
@@ -170,6 +170,10 @@ rp1_target: &pcie2 {
 	status = "okay";
 };
 
+&pcie1 {
+	brcm,vdm-qos-map = <0x33333333>;
+};
+
 // Add some labels to 2712 device
 
 // The system UART
